### DISCUSSION TITLE
help needed: bootlooping after upgrade and toltec reenable

### DISCUSSION THREAD

**Discussion by wordpr3ss** [toltec-dev/toltec#671] [Support] [answered]
After upgrading to the latest firmware and reenabling toltec (didn't remember any errors when doing so), the device started bootlooping: showing "starting" ->  a long while -> screen flashes -> showing "restarting" -> a short while -> screen flashes -> loopback.

Don't remember much detail.

Current trying https://github.com/ddvk/remarkable2-recovery , but couldn't get anything on the dmesg log on the host wrt to the device.

Any suggestion to unbrick the device?
Thanks!

**Comment by LinusCDE** [comment]
What launcher were you using last (if any)?

I'm not aware that rm2fb or any launcher initiates a reboot if failing to start to my knowledge. Maybe some watchdog triggering it?

If it takes that long to boot, I'd assume that SSH should be up for quite a while and it's just failing to display a third party launcher due to out-of-date [rm2fb](https://github.com/ddvk/remarkable2-framebuffer/). Creating a file called `/opt/etc/rm2fb.conf` and inserting the correct/newest snippet from [this issue](https://github.com/ddvk/remarkable2-framebuffer/issues/18) might resolve it.

Otherwise you could try to revert to the pre-updated rootfs. I think I found this script in remarkable-hacks at some point but can't find it there anymore, so here it is:
```bash
#!/bin/bash
# switches the active root partition

fw_setenv "upgrade_available" "1"
fw_setenv "bootcount" "0"

OLDPART=$(fw_printenv -n active_partition)
if [ $OLDPART  ==  "2" ]; then
    NEWPART="3"
else
    NEWPART="2"
fi
echo "new: ${NEWPART}"
echo "fallback: ${OLDPART}"

fw_setenv "fallback_partition" "${OLDPART}"
fw_setenv "active_partition" "${NEWPART}"
```

SSH into the device, run `nano switch.sh` paste the above, exit nano with CTRL+O, Enter, CTRL+X and run `chmod +x switch.sh` and finally `./switch.sh` to execute it.

Edit: SSH will most likely not be available over WiFi, but it should hopefully be over USB. If you connect the remarkable to your PC you should be able to ssh into it using the IP `10.11.99.1`.

**Comment by Eeems** [comment, marked as answer]
The latest firmware is not supported, see the warning on the website: https://toltec-dev.org/#install-toltec

The FAQs there also talk about how to recover from an unresponsive UI if you still have SSH access: https://toltec-dev.org/#frequently-asked-questions

For help with remarkable2-recovery, please ask in that repo, as this is for toltec itself. You can also get help on the community discord.

**Comment by wordpr3ss** [comment]
at the time the last was supported. I waited a couple of months for the recovery parts to come (listed in the recovery repo)

There is no ssh access to the device any more. neither usb nor wifi.

so I couldn't quote which version.

**Reply by Eeems** [reply]
Since you don't have SSH access, you will be needing help with remarkable2-recovery, which you can find on that repository, or in the community discord.
